src/ess_rhat.jl

```diff
@@ -233,25 +236,206 @@ The ESS and ``\\widehat{R}`` values can be computed for the following estimators
+- `:bulk`/`:rank`: mean-ESS computed on rank-normalized draws. This type diagnoses poor
+    convergence in the bulk of the distribution due to trends or different locations of the
+    chains.
+- `:tail`: minimum of the quantile-ESS for the symmetric quantiles where
+    `tail_prob=0.1` is the probability in the tails. This type diagnoses poor convergence in
+    the tails of the distribution. If this type is chosen, `kwargs` may contain a
+    `tail_prob` keyword.
+- `:basic`: basic ESS, equivalent to specifying `estimator=Statistics.mean`.
+
+While Bulk-ESS is conceptually related to basic ESS, it is well-defined even if the chains
+do not have finite variance.[^VehtariGelman2021]. For each parameter, rank-normalization
+proceeds by first ranking the inputs using "tied ranking" and then transforming the ranks to
+normal quantiles so that the result is standard normally distributed. This transform is
+monotonic.
+
 [^VehtariGelman2021]: Vehtari, A., Gelman, A., Simpson, D., Carpenter, B., & Bürkner, P. C. (2021).
     Rank-normalization, folding, and localization: An improved ``\\widehat {R}`` for
     assessing convergence of MCMC. Bayesian Analysis.
     doi: [10.1214/20-BA1221](https://doi.org/10.1214/20-BA1221)
     arXiv: [1903.08008](https://arxiv.org/abs/1903.08008)
 """
-function ess_rhat(samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
-    return ess_rhat(Statistics.mean, samples; kwargs...)
+@constprop :aggressive function ess(
+    samples::AbstractArray{<:Union{Missing,Real},3};
+    estimator=nothing,
+    type=nothing,
+    kwargs...,
+)
+    if estimator !== nothing && type !== nothing
+        throw(ArgumentError("only one of `estimator` and `type` can be specified"))
+    elseif estimator !== nothing
+        return _ess(estimator, samples; kwargs...)
+    elseif type !== nothing
+        return _ess(_val(type), samples; kwargs...)
+    else
+        return _ess(Val(:basic), samples; kwargs...)
+    end
 end
-function ess_rhat(f, samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
-    x = _expectand_proxy(f, samples)
+function _ess(estimator, samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
+    x = _expectand_proxy(estimator, samples)
     if x === nothing
-        throw(ArgumentError("the estimator $f is not yet supported by `ess_rhat`"))
+        throw(ArgumentError("the estimator $estimator is not yet supported by `ess`"))
+    end
+    return _ess(Val(:basic), x; kwargs...)
+end
+function _ess(::Val{T}, ::AbstractArray{<:Union{Missing,Real},3}; kwargs...) where {T}
+    return throw(ArgumentError("the `type` `$T` is not supported by `ess`"))
+end
+function _ess(type::Val{:basic}, samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
+    return first(_ess_rhat(type, samples; kwargs...))
+end
+function _ess(type::Val{:bulk}, samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
+    return first(_ess_rhat(type, samples; kwargs...))
+end
```

Comment:
Couldn't we just use
```suggestion
function _ess(type::Val, samples::AbstractArray{<:Union{Missing,Real},3}; kwargs...)
    return first(_ess_rhat(type, samples; kwargs...))
end
```
and error if `_ess_rhat` does not support the `type`?